pyccel/parser/semantic.py

```diff
@@ -3275,6 +3275,8 @@ def _visit_Return(self, expr):
             if not (isinstance(r, PyccelSymbol) and r == (v.name if isinstance(v, Variable) else v)):
                 a = self._visit(Assign(v, r, fst=expr.fst))
                 assigns.append(a)
+                if isinstance(a, ConstructorCall):
+                    a.cls_variable.is_temp = False
```

Comment:
Can you explain this change? Does the variable need an explicit declaration in every language? Can you show me the C code so I can see why this is necessary please